test/regression/solvers/delta_eddington.F90

```diff
@@ -0,0 +1,356 @@
+    use tuvx_core,                     only: core_t
+    use tuvx_grid,                     only: grid_t
+    use tuvx_profile,                  only: profile_t
+    use tuvx_radiator,                 only: radiator_state_t
+    use tuvx_solver,                   only: radiation_field_t
+
+    type(core_t),             intent(in) :: tuvx_core
+    real(dk),                 intent(in) :: solar_zenith_angles(:)
+    real(dk),                 intent(in) :: earth_sun_distances(:)
+    type(radiator_state_t),   intent(in) :: radiator_states(:)
+    type(radiation_field_t), allocatable :: radiation_fields(:)
+
+    class(grid_t), pointer :: heights
+    class(grid_t), pointer :: wavelengths
+    real(kind=c_double), allocatable, target :: solar_zenith_angles_c(:)
+    real(kind=c_double), allocatable, target :: earth_sun_distances_c(:)
+    real(kind=c_double), allocatable, target :: altitude_mid_points_c(:,:)
+    real(kind=c_double), allocatable, target :: altitude_edges_c(:,:)
+    real(kind=c_double), allocatable, target :: wavelength_mid_points_c(:)
+    real(kind=c_double), allocatable, target :: wavelength_edges_c(:)
+    real(kind=c_double), allocatable, target :: layer_OD_c(:,:,:)
+    real(kind=c_double), allocatable, target :: layer_SSA_c(:,:,:)
+    real(kind=c_double), allocatable, target :: layer_G_c(:,:,:)
+    type(solver_input_t_c) :: input
+    type(solver_output_t_c) :: output
+    integer :: i_col, n_col, n_lev, n_wl
+
+    ! Set up the input struct
+    heights => tuvx_core%get_grid( "height", "km" )
+    wavelengths => tuvx_core%get_grid( "wavelength", "nm" )
+    input%n_wavelengths_ = wavelengths%ncells_
+    input%n_levels_      = heights%ncells_
+    input%n_columns_     = size( solar_zenith_angles )
+    solar_zenith_angles_c   = real( solar_zenith_angles(:), kind=c_double )   &
+                                  * real(pi, kind=c_double) / 180.0_c_double ! degrees -> radians
+    earth_sun_distances_c   = real( earth_sun_distances(:), kind=c_double )
+    allocate( altitude_mid_points_c( size( solar_zenith_angles ),             &
+                                     heights%ncells_ ) )
+    allocate( altitude_edges_c(      size( solar_zenith_angles ),             &
+                                     heights%ncells_+1 ) )
+    allocate( layer_OD_c(            size( solar_zenith_angles ),             &
+                                     heights%ncells_, wavelengths%ncells_ ) )
+    allocate( layer_SSA_c(           size( solar_zenith_angles ),             &
+                                     heights%ncells_, wavelengths%ncells_ ) )
+    allocate( layer_G_c(             size( solar_zenith_angles ),             &
+                                     heights%ncells_, wavelengths%ncells_ ) )
+    do i_col = 1, size( solar_zenith_angles )
+      altitude_mid_points_c(i_col,:) =                                        &
+          real( heights%mid_(:), kind=c_double ) * 1.0e3_c_double ! km -> m
```

Comment:
why do we have to do this unit conversion here? I thought both solvers operated on the same units